storage/persistence_test.go

```diff
@@ -0,0 +1,32 @@
+package storage
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/evergreen-ci/logkeeper/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUploadBuildMetadata(t *testing.T) {
+	storage := makeTestStorage(t, "")
+	defer os.RemoveAll(tempDir)
```

Comment:
Not sure if we want to do this in this change, but we may want to make a separate cleanup function for the test storage so that it's at the same level of abstraction as "makeTestStorage" (that is we're not just removing tempDir in all of our functions).

Reply:
This is more important if we use an S3 bucket optionally in tests. We can have a "cleanupTestStorage" function that knows how to clear the temporary directory. I tried using the bucket's remove function but that only deletes files and not directories if you are using a filesystem based storage.